### PR TITLE
Devtools: introduces a platform release index allowing proper platform releases ordering when selecting BOMs for import

### PR DESCRIPTION
**Pull request by aloubyansky**
This PR allows to properly prioritize platforms for a given selection of extensions and select the latest BOMs recommended by the registries.

### CODE REVIEW

**Comment by aloubyansky** [comment]
@ia3andy or @maxandersen could at least one of you review this one please? I would like to have this in 2.0.0.Final.

**Comment by aloubyansky** [comment]
With this change, assuming the registry returns a platform catalog like
````
{
  "platforms" : [ {
    "platform-key" : "io.quarkus.platform",
    "streams" : [ {
      "id" : "999-SNAPSHOT",
      "releases" : [ {
        "version" : "0",
        "member-boms" : [ "io.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-hazelcast-client-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-optaplanner-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-qpid-jms-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-blaze-persistence-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-debezium-outbox-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT", "io.quarkus.platform:quarkus-kogito-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT" ],
        "quarkus-core-version" : "2.0.0.CR2"
      } ]
    }, {
      "id" : "1.13",
      "releases" : [ {
        "version" : "5",
        "member-boms" : [ "io.quarkus:quarkus-universe-bom-quarkus-platform-descriptor:1.13.5.Final:json:1.13.5.Final" ],
        "quarkus-core-version" : "1.13.5.Final"
      } ]
    } ]
  } ]
}
````

The CLI would create a project for command

`quarkus create cassandra-quarkus-client --registry-client`

importing `io.quarkus:quarkus-universe-bom:1.13.5.Final`, given that it's the last platform release including the Cassandra extension (quite a few extensions won't make it into 2.0.0.Final).

And for `quarkus -e create kogito-quarkus-rules --registry-client`, assuming Kogito is included in 2.0.0.Final, the project will be importing `io.quarkus.platform:quarkus-bom:999-SNAPSHOT` and `io.quarkus.platform:quarkus-kogito-bom:999-SNAPSHOT`.

This will not require a user to search for the proper stream in which a specific extension is available.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java [hunk 1]
```diff
@@ -79,43 +70,46 @@ public QuarkusCommandOutcome execute(QuarkusCommandInvocation invocation) throws
...
-            final Iterator<ArtifactCoords> i = boms.iterator();
-            while (i.hasNext()) {
-                final ArtifactCoords next = i.next();
-                // TODO we remove quarkus-bom here because it's currently added by default using properties in the template
-                // ideally, it shouldn't be different than the rest of the imported BOMs
-                if (next.getArtifactId().equals("quarkus-bom")) {
-                    bom = next;
-                    i.remove();
-                    break;
+        ExtensionCatalog mainPlatform = invocation.getExtensionsCatalog(); // legacy platform initialization
```
Comment (ia3andy) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:73]: didn't we move that to a method?

Comment (ia3andy) [reply independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/CreateProjectCommandHandler.java:73]: Aw sorry it's another, maybe this should also be in a method then :)

**Review by ia3andy — APPROVED** [review]
LGTM

**Pull request merged**